docs/user/Contradictions.md

```diff
@@ -1,43 +1,46 @@
 # Contradictions
 
-Contradictions are where constraints are combined in some fashion, creating a representation of data where no, or incorrectly reduced, amounts of data can be produced. The following categories of contradictions can occur in a profile:
+Contradictions are where constraints are combined in some fashion, creating a representation of data where no, or incorrectly reduced, amounts of data can be produced.
+The following categories of contradictions can occur in a profile:
 
-This document describes [how data is generated](SetRestrictionAndGeneration.md) and underpins the concept of contradictions.
 
 | Type | Explanation | Example |
 | ---- | ---- | ---- |
 | 'Hard' | _cannot_ create any rows of data (for the field, and therefore the output file) | `foo ofType string` and `foo ofType decimal` and `foo not(is null)` - no rows would be emitted |
-| 'Soft' | _could_ create some data, but some scenarios would produce none | `foo not null` and `if bar equalTo 1 then foo is null`, also `foo ofType string` and `foo ofType decimal`, 1 row would be emitted, containing `null` |
+| 'Partial' | _could_ create some data, but some scenarios would produce none | `foo equalTo 1` and `anyOf foo equalTo 1 or foo equalTo 2` (1 is produced), also `foo ofType string` and `foo ofType decimal` (`null` is produced) |
+
+Note that the phases "there is a hard contradiction", "the profile is fully contradictory" and "the profile is wholly contradictory" are synonymous.
 
-There is an optional component of the generator - the profile validator - which can detect some of the above contradictions. See the explanation of each type for more detail.
+This document describes [how data is generated](SetRestrictionAndGeneration.md) and underpins the concept of contradictions.
 
 ## 'Hard' contradictions
 This is where contradictions occur in a way where no rows could be satisfied for at least one field. If this is the case then no rows can be emitted for any field in the profile, therefore the output file/s would be empty.
-Hard contradictions can otherwise be described as removing all possible values from the universal set and denying the absence of values for the field; `not(is null)`. All hard contradictions must contain this constraint in some fashion, otherwise the field can still have no value (regularly represented as `null`).
+Hard contradictions can otherwise be described as removing all possible values from the universal set and denying the absence of values for the field; `not(is null)`.
+
+Note that these contradictions can occur even if most of the profile is correct. If no data can be generated for a single field,
+then it will prevent all the other fields from producing data.
```

Comment:
I think there is an extra line break here.

> Note that these contradictions can occur even if most of the profile is correct. If no data can be generated for a single field,
then it will prevent all the other fields from producing data.

Should be:

>Note that these contradictions can occur even if most of the profile is correct. If no data can be generated for a single field, then it will prevent all the other fields from producing data.